scripts/create-instance.sh

```diff
@@ -23,77 +23,77 @@
     echo "[ERROR] Example: export INVENIO_WEB_VENV=invenio3"
     exit 1
 fi
-if [ -v ${INVENIO_USER_EMAIL} ]; then
+if [[ -v INVENIO_USER_EMAIL ]]; then
     echo "[ERROR] Please set environment variable INVENIO_USER_EMAIL before runnning this script."
     echo "[ERROR] Example: export INVENIO_USER_EMAIL=info@inveniosoftware.org"
     exit 1
 fi
-if [ -v ${INVENIO_USER_PASS} ]; then
+if [[ -v INVENIO_USER_PASS ]]; then
     echo "[ERROR] Please set environment variable INVENIO_USER_PASS before runnning this script."
     echo "[ERROR] Example: export INVENIO_USER_PASS=uspass123"
     exit 1
 fi
-if [ -v ${INVENIO_POSTGRESQL_HOST} ]; then
+if [[ -v INVENIO_POSTGRESQL_HOST ]]; then
     echo "[ERROR] Please set environment variable INVENIO_POSTGRESQL_HOST before runnning this script."
     echo "[ERROR] Example: export INVENIO_POSTGRESQL_HOST=192.168.50.11"
     exit 1
 fi
-if [ -v ${INVENIO_POSTGRESQL_DBNAME} ]; then
+if [[ -v INVENIO_POSTGRESQL_DBNAME ]]; then
     echo "[ERROR] Please set environment variable INVENIO_POSTGRESQL_DBNAME before runnning this script."
     echo "[ERROR] Example: INVENIO_POSTGRESQL_DBNAME=invenio3"
     exit 1
 fi
-if [ -v ${INVENIO_POSTGRESQL_DBUSER} ]; then
+if [[ -v INVENIO_POSTGRESQL_DBUSER ]]; then
     echo "[ERROR] Please set environment variable INVENIO_POSTGRESQL_DBUSER before runnning this script."
     echo "[ERROR] Example: INVENIO_POSTGRESQL_DBUSER=invenio3"
     exit 1
 fi
-if [ -v ${INVENIO_POSTGRESQL_DBPASS} ]; then
+if [[ -v INVENIO_POSTGRESQL_DBPASS ]]; then
     echo "[ERROR] Please set environment variable INVENIO_POSTGRESQL_DBPASS before runnning this script."
     echo "[ERROR] Example: INVENIO_POSTGRESQL_DBPASS=dbpass123"
     exit 1
 fi
-if [ -v ${INVENIO_REDIS_HOST} ]; then
+if [[ -v INVENIO_REDIS_HOST ]]; then
     echo "[ERROR] Please set environment variable INVENIO_REDIS_HOST before runnning this script."
     echo "[ERROR] Example: export INVENIO_REDIS_HOST=192.168.50.12"
     exit 1
 fi
-if [ -v ${INVENIO_ELASTICSEARCH_HOST} ]; then
+if [[ -v INVENIO_ELASTICSEARCH_HOST ]]; then
     echo "[ERROR] Please set environment variable INVENIO_ELASTICSEARCH_HOST before runnning this script."
     echo "[ERROR] Example: export INVENIO_ELASTICSEARCH_HOST=192.168.50.13"
     exit 1
 fi
-if [ -v ${INVENIO_RABBITMQ_HOST} ]; then
+if [[ -v INVENIO_RABBITMQ_HOST ]]; then
     echo "[ERROR] Please set environment variable INVENIO_RABBITMQ_HOST before runnning this script."
     echo "[ERROR] Example: export INVENIO_RABBITMQ_HOST=192.168.50.14"
     exit 1
 fi
-if [ -v ${INVENIO_WORKER_HOST} ]; then
+if [[ -v INVENIO_WORKER_HOST ]]; then
     echo "[ERROR] Please set environment variable INVENIO_WORKER_HOST before runnning this script."
     echo "[ERROR] Example: export INVENIO_WORKER_HOST=192.168.50.15"
     exit 1
 fi
 
 # load virtualenvrapper:
-source $(which virtualenvwrapper.sh)
+source virtualenvwrapper.sh
```

Comment:
Regarding `source $(which virtualenvwrapper.sh)` and removing the `which` part, IIRC this was necessary in the previous Invenio 1/2 kickstart scripts depending on the presence of the system package, the venv package, the possible `--system-site-packages` option, and whether the venv was activated or not before running those `source` lines (in the same shell session).  

It was safer to use `which` as part of the command for some of these combinations. Here it may be safer.  Just wanted to check whether you tested both Vagrant and Docker installation techniques to see whether it can now be safely removed? 

In case of doubt, it's safer to leave the `which` bit there.

P.S. We have some more occurrences of `which` in the other scripts.